### PR TITLE
[FLINK-6513][docs] cleaned up some typos and grammatical flaws

### PR DESCRIPTION
**Pull request by alpinegizmo**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

### CODE REVIEW

**Comment by zentol** [comment]
+1 to merge.

File: docs/dev/stream/checkpointing.md [hunk 1]
```diff
@@ -124,8 +124,8 @@ env.getCheckpointConfig.setMaxConcurrentCheckpoints(1)
 
 ## Selecting a State Backend
 
-The checkpointing mechanism stores the progress in the data sources and data sinks, the state of windows, as well as the [user-defined state](state.html) consistently to
-provide *exactly once* processing semantics. Where the checkpoints are stored (e.g., JobManager memory, file system, database) depends on the configured
+The checkpointing mechanism stores consistent snapshots of the progress in the data sources and data sinks, the state of windows and timers, as well as any [user-defined state](state.html), thereby
```
Comment (greghogan) [inline docs/dev/stream/checkpointing.md]: Perhaps this first sentence could use a little more work. Can we better explain "progress"? "as well as" -> "and"? Could we move "providing *exactly once* processing semantics" to the beginning: "The checkpointing mechanism provides ..."?

File: docs/dev/best_practices.md [hunk 2]
```diff
@@ -284,30 +284,30 @@ Change your projects `pom.xml` file like this:
...
+ * Exclude all `log4j` dependencies from all Flink dependencies: this causes Maven to ignore Flink's transitive dependencies to log4j.
+ * Exclude the `slf4j-log4j12` artifact from Flink's dependencies: since we are going to use the slf4j to logback binding, we have to remove the slf4j to log4j binding.
  * Add the Logback dependencies: `logback-core` and `logback-classic`
- * Add dependencies for `log4j-over-slf4j`. `log4j-over-slf4j` is a tool which allows legacy applications which are directly using the Log4j APIs to use the Slf4j interface. Flink depends on Hadoop which is directly using Log4j for logging. Therefore, we need to redirect all logger calls from Log4j to Slf4j which is in turn logging to Logback.
+ * Add dependencies for `log4j-over-slf4j`. `log4j-over-slf4j` is a tool which allows legacy applications which are directly using the Log4j APIs to use the Slf4j interface. Flink depends on Hadoop which is directly using Log4j for logging. Therefore we need to redirect all logger calls from Log4j to Slf4j which is in turn logging to Logback.
```
Comment (alpinegizmo) [inline docs/dev/best_practices.md]: That was a mistake.

File: docs/dev/stream/checkpointing.md (hunk 1 above)
Comment (alpinegizmo) [inline docs/dev/stream/checkpointing.md]: After spending more time with this, my conclusion is that getting into "exactly once processing semantics" is off-topic in a short section whose purpose is to introduce the idea of state backends. What I've done instead is to beef up the linking between the various related pages.

**Comment by greghogan** [comment]
Merging ...

**Pull request closed without merging**